### PR TITLE
Borgbackup 1.1.2

### PR DESCRIPTION
**Pull request by yuroller**
_Motivation:_ borgbackup v.1.1.2 package.
_Linked issues:_ https://github.com/SynoCommunity/spksrc/pull/2906

### Checklist
- [] Build rule `all-supported` completed successfully
- [] Package upgrade completed successfully
- [*] New installation of package completed successfully

### CODE REVIEW

File: cross/acl/Makefile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+PKG_EXT = tar.gz
+PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).src.$(PKG_EXT)
+PKG_DIST_SITE = http://download.savannah.gnu.org/releases/acl
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
+
+DEPENDS = cross/attr
+
+HOMEPAGE = http://kernel.org/
+COMMENT  = 
+LICENSE  =
```
Comment (cytec) [inline cross/acl/Makefile]: should be GPLv2+ as of https://savannah.nongnu.org/projects/acl
Note: might add the Comment there as well ;)

File: cross/borgbackup/Makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+PKG_EXT = tar.gz
+PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = https://files.pythonhosted.org/packages/source/b/$(PKG_NAME)
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
+
+DEPENDS = cross/acl cross/lz4 cross/openssl
+
+HOMEPAGE = https://borgbackup.readthedocs.io/
+COMMENT  = Deduplicating backup program with compression and authenticated encryption
+LICENSE  = Borg
```
Comment (cytec) [inline cross/borgbackup/Makefile]: isn't Borg BSD?

File: cross/lz4/Makefile [hunk 3]
```diff
@@ -0,0 +1,22 @@
+PKG_NAME = lz4
+PKG_VERS = 1.7.5
+PKG_EXT = tar.gz
+PKG_DIST_NAME = v$(PKG_VERS).$(PKG_EXT)
```
Comment (cytec) [inline cross/lz4/Makefile]: Please use`PKG_DIST_FILE` for a better tar.gz name. example see: https://github.com/SynoCommunity/spksrc/blob/master/cross/rtmpdump/Makefile#L5
basically something like `$(PKG_NAME )-v$(PKG_VERS).$(PKG_EXT)` should be good ;)

File: cross/acl/Makefile (hunk 1 above)
Comment (yuroller) [reply cross/acl/Makefile]: Ok, I will update License and add Comment for acl package

File: cross/borgbackup/Makefile (hunk 2 above)
Comment (yuroller) [reply cross/borgbackup/Makefile]: I do not know: licence is this https://github.com/borgbackup/borg/blob/master/LICENSE
What shall I put in the Makefile?

File: cross/lz4/Makefile (hunk 3 above)
Comment (yuroller) [reply cross/lz4/Makefile]: Ok, I will try to use PKG_DIST_FILE for lz4 package

**Comment by cytec** [comment]
@yuroller normally i'd say: 1 PR per package. So python and borg would be one PR as both is needed for borg to work, the typo fix in an additional one. since it's only a typo that's been fixed i'd be ok with it to have it in this one for now. don't know about others \cc @SynoCommunity/developers

File: cross/borgbackup/Makefile (hunk 2 above)
Comment (cytec) [reply cross/borgbackup/Makefile]: hm.. seems like on here it's mentioned to be BSD 3 Clause.
https://github.com/borgbackup/borg#main-features (Free and Open Source Software)

**Comment by ymartin59** [comment]
Republishing Python 3 package for some native packages would worth for an complete update to 3.6.3 ?

**Comment by cytec** [comment]
agree if we can push python to 3.6.3 that'll be a seperate PR for just python then :)

**Comment by yuroller** [comment]
Please, update python3 package, then I will do a PR for borgbackup with modification needed in python3 for missing modules

**Comment by Safihre** [comment]
@yuroller Are you still willing to work on this? I am trying to understand this PR, since there already is a `borgbackup` package with these cross dependencies.
Would it now only be needed to bump the version to 1.1.4 (latest)?
Seems though that even though the package is present here, it's not published on the repo.
Is this due to DSM6 compatibility @ymartin59 ?

**Comment by ymartin59** [comment]
@Safihre No idea. I just found borgbackup package has been merged from #2694 but probably never published.

**Comment by yuroller** [comment]
On the weekend I try to find the time to submit a new pull request to master branch (single commit if I remember well),

**Comment by Safihre** [comment]
Great, looking forward. Would you also be able to convert it to Generic Service approach?

**Comment by yuroller** [comment]
What is "generic service" approach?
I created synology borg package by copying other similar python/c already present in spksrc repository, so if you point me to an example I can try doing the "generic service".
BTW borg is a command line program, not a daemon/service, so I do not know if that applies.

**Comment by Safihre** [comment]
In that case you can look at Midnight Commander (`mc`). That's very similar.

I looked at this package and was bit surprised that all the modules were added to the base `python3` and not as part of just this package. That should easily be possible to keep the Python packages clean.
But that's something I can do afterwards :)

**Comment by yuroller** [comment]
I did a new pull request with updated borgbackup https://github.com/SynoCommunity/spksrc/pull/3196
I had to add cross/attr and cross/acl to python3 package or else borgbackup is not able to find the .so if I include them just in borgbackup package: can someone explain me why?
About "generic service": the .sh scripts I use are very similar to those in "mc" package so it should be fine.

**Comment by Safihre** [comment]
Let's continue discussions in the new PR then :)
Will look into this Python thing and report back in new PR!

**Pull request closed without merging**